### ISSUE TITLE
ECMAScript module loading error

### ISSUE THREAD

**Issue by evana-kpo** [jaydenseric/apollo-upload-server#40]
Thank you for your great package. I have one question:

When using the program with ESM (.mjs import) I get an error with the current version of Node (9.4.0).

```javascript
(node:4290) ExperimentalWarning: The ESM module loader is experimental.
file:///Users/.../node_modules/apollo-upload-server/lib/types.mjs:1
import { GraphQLScalarType } from 'graphql'
         ^^^^^^^^^^^^^^^^^
SyntaxError: The requested module does not provide an export named 'GraphQLScalarType'
    at ModuleJob._instantiate (internal/loader/ModuleJob.js:86:19)
    at <anonymous>
error Command failed with exit code 1.
```

The problem should be resolved by changing the following lines in `types.mjs`:

```javascript
import graphql from 'graphql'

const { GraphQLScalarType } = graphql
```

I think GraphQL exports currently only with CJS and not yet with ESM and therefore there is only the "default export". Since you are running this program with Babel, you will not get an error. I use the package directly with Node.

https://nodejs.org/api/esm.html#esm_interop_with_existing_modules

**Comment by jaydenseric** [comment]
I was using vanilla `--experimental-modules`, but have recently been using [`@std/esm`](https://github.com/standard-things/esm) in `cjs` mode so it's possible this slipped under the radar. The intent is to be 100% vanilla .mjs compatible. Looking into it now…

**Comment by evana-kpo** [comment]
I started the program also with `node --experimental-modules ./server.mjs` and `@std/esm`, but no success. The same without `@std/esm`. Only the error message is more informative. I found no solution other than import by default and destructuring the imported default object. Thank you for your help.

**Comment by jaydenseric** [comment]
Hang on, the fix is not working 😱

**Comment by jaydenseric** [comment]
The issue is that this in the source works transpiles correctly for .mjs, but breaks in the .js:

```js
import graphql from 'graphql'
export const GraphQLUpload = new graphql.GraphQLScalarType({ … })
```

While this breaks the .mjs but works for the .js:

```js
import * as graphql from 'graphql'
export const GraphQLUpload = new graphql.GraphQLScalarType({ … })
```

Not sure if it is a Babel bug or what.

**Comment by evana-kpo** [comment]
Ok, maybe you can use a temporary solution with a mix of default and variable import like this one:
`import myDefault, * as myModule from "my-module";`
Not sure if it works for both, vanilla node and babel.

**Comment by jaydenseric** [comment]
Related? https://github.com/babel/babel/issues/6983

**Comment by jaydenseric** [comment]
This should be fixed in [v4.0.2](https://github.com/jaydenseric/apollo-upload-server/releases/tag/v4.0.2). I'll continue to investigate a better solution.

I've updated [the example API](https://github.com/jaydenseric/apollo-upload-examples/tree/master/api) to use native modules instead of `@std/esm`.

**Comment by evana-kpo** [comment]
Now, it works - thanks 👍
